CarDealershipSystem.java

```diff
@@ -1,302 +1,291 @@
+	public List<Customer> customers = new ArrayList<>();
+	public List<Car> cars = new ArrayList<>();
+	public List<Transaction> transactions = new ArrayList<>();
+	public List<Car> sold = new ArrayList<>();
+
+	public boolean authenticate(String assID, String assPass) {
+
+		for (SalesAssociate associate : salesAssociates) {
+			if (associate.getAssociateID().equals(assID) && associate.getAssociatePassword().equals(assPass)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public void loadFiles() throws IOException {
+		loadCars();
+		loadSalesAssociates();
+		loadTransactions();
+		loadCustomers();
+		loadSold();
+	}
+
+	private void loadCars() throws IOException {
+		File testFile = new File("E:/Users/johnn/Documents/cars.txt");
+		BufferedReader reader = new BufferedReader(new FileReader(testFile));
+		String currentLine;
+
+		while ((currentLine = reader.readLine()) != null) {
+
+			Car car = new Car(currentLine);
+			cars.add(car);
+		}
+		reader.close();
+	}
+
+	private void loadSalesAssociates() throws IOException {
+		File testFile = new File("E:/Users/johnn/Documents/employees.txt");
+		BufferedReader reader = new BufferedReader(new FileReader(testFile));
+		String currentLine;
+
+		while ((currentLine = reader.readLine()) != null) {
+
+			SalesAssociate associate = new SalesAssociate(currentLine);
+			salesAssociates.add(associate);
+		}
+		reader.close();
+	}
+
+	private void loadTransactions() throws IOException {
+		File testFile = new File("E:/Users/johnn/Documents/transactions.txt");
+		BufferedReader reader = new BufferedReader(new FileReader(testFile));
+		String currentLine;
+
+		while ((currentLine = reader.readLine()) != null) {
+
+			Transaction transaction = new Transaction(currentLine);
+			transactions.add(transaction);
+		}
+		reader.close();
+	}
+
+	private void loadCustomers() throws IOException {
+		File testFile = new File("E:/Users/johnn/Documents/customers.txt");
+		BufferedReader reader = new BufferedReader(new FileReader(testFile));
+		String currentLine;
+
+		while ((currentLine = reader.readLine()) != null) {
+
+			Customer customer = new Customer(currentLine);
+			customers.add(customer);
+		}
+		reader.close();
+	}
+
+	private void loadSold() throws IOException {
+		File testFile = new File("E:/Users/johnn/Documents/sold.txt");
+		BufferedReader reader = new BufferedReader(new FileReader(testFile));
+		String currentLine;
+
+		while ((currentLine = reader.readLine()) != null) {
+
+			Car car = new Car(currentLine);
+			sold.add(car);
+		}
+		reader.close();
+	}
+
+	public void addCar(String make, String model, String vin, String year, String color, double value)
+			throws IOException {
+		Car car = new Car(make, model, vin, year, color, value);
+		cars.add(car);
+	}
+
+	public Car findCar(String vin) {
+		for (Car car : cars) {
+			if (car.getCarVIN().equals(vin)) {
+				return car;
+			}
+		}
+		return null;
+	}
+
+	public List<Car> findCars(String make, String model) { // figure out multiples

```

Comment:
In general, you should return the interface, not the implementation.

Reply:
thank you